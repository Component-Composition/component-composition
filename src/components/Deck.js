import React from 'react';
import PropTypes from 'prop-types';

const Deck = ({ children, title }) => {
  return (
    <section>
      {title && <header>{title}</header>}
      {children}
    </section>
  );
};

Deck.propTypes = {
  children: PropTypes.node.isRequired,
  title: PropTypes.string
};

export default Deck;
