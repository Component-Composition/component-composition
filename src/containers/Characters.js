import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Cards from '../components/DeckOfCards';

export default class Characters extends Component {
  constructor(props) {
    super(props);
  
    this.state = {
      data: []
    };
  }

  static propTypes = {
    count: PropTypes.number
  }

  static defaultProps = {
    count: 10
  }

  componentDidMount() {
    const URL = `https://hey-arnold-api.herokuapp.com/api/v1/characters?perPage=${this.props.count}`;

    fetch(URL)
      .then(response => response.json())
      .then(data => this.setState({ data }));
  }

  render() {
    const { data } = this.state;
    return (
      <Cards items={data} />
    );
  }
}
