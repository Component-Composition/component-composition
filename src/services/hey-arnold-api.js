function getItems(url) {
  return fetch(url)
    .then(response => response.json());
}

export { getItems };
