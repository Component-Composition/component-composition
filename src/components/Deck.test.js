import React from 'react';
import { shallow } from 'enzyme';
import Deck from './Deck';
import Characters from '../containers/Characters';
import { getItems } from '../services/hey-arnold-api';

jest.mock('../services/hey-arnold-api.js', () => {
  return {
    getItems() {
      return Promise.resolve([{
        '_id': '5da237699734fdcb7bef8f51',
        'name': 'Arnold Shortman',
        'image': 'https://vignette.wikia.nocookie.net/heyarnold/images/f/f5/Arnold.png/revision/latest?cb=20181027162333'
      }]);
    }
  };
});

describe('Deck Component', () => {
  it('renders a deck', () => {
    return getItems().then(item => {
      const wrapper = shallow(<Deck><Characters items={item}  /></Deck>);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
