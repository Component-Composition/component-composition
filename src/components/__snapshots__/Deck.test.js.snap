// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deck Component renders a deck 1`] = `
<section>
  <Characters
    count={10}
    items={
      Array [
        Object {
          "_id": "5da237699734fdcb7bef8f51",
          "image": "https://vignette.wikia.nocookie.net/heyarnold/images/f/f5/Arnold.png/revision/latest?cb=20181027162333",
          "name": "Arnold Shortman",
        },
      ]
    }
  />
</section>
`;
